src/target/target_probes.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * This file is part of the Black Magic Debug project.
+ *
+ * Copyright (C) 2022 Rafael Silva <perigoso@riseup.net>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "target_probes.h"
+
+#define CORTEXA_PROBE_DUMMY __attribute__((weak, alias("_cortexa_probe_dummy")))
+#define CORTEXM_PROBE_DUMMY __attribute__((weak, alias("_cortexm_probe_dummy")))
+#define TARGET_PROBE_DUMMY __attribute__((weak, alias("_target_probe_dummy")))
+
+static inline bool _target_probe_dummy(target *t)
```

Comment:
Rather than calling these "dummy", and as the _ on the front makes them reserved naming violating, we'd prefer, eg, `target_probe_nop` or similar.

Reply:
nop does is better :+1: